allennlp_models/generation/models/simple_seq2seq.py

```diff
@@ -367,7 +367,11 @@ def _forward_loop(
         step_logits: List[torch.Tensor] = []
         step_predictions: List[torch.Tensor] = []
         for timestep in range(num_decoding_steps):
-            if self.training and torch.rand(1).item() < self._scheduled_sampling_ratio:
+            if (
+                self.training
+                and self._scheduled_sampling_ratio > 0.0
+                and torch.rand(1).item() < self._scheduled_sampling_ratio
+            ):
```

Comment:
@epwalsh Added a similar condition to `simple_seq2seq` to avoid the call to `torch.rand` when `_scheduled_sampling_ratio` is `0.0`.